pandas/tests/reshape/concat/test_concat.py

```diff
@@ -445,6 +446,54 @@ def test_concat_ordered_dict(self):
         tm.assert_series_equal(result, expected)
 
 
+@pytest.mark.parametrize("index_maker", tm.index_subclass_makers_generator())
+@pytest.mark.parametrize("join", ["inner", "outer"])
+def test_concat_duplicates_error(index_maker, join):
+    # if index_maker is tm.makeMultiIndex:
+    # TODO: This generator only makes Indexs of size 4
+    # pytest.skip()
+    index_unique = index_maker(k=4)
+    index_non_unique = index_unique[[0, 0, 1, 2, 3]]
+
+    df_non_unique = pd.DataFrame(
+        np.ones((1, len(index_non_unique))), columns=index_non_unique
+    )
+    df_unique = pd.DataFrame(np.ones((1, len(index_unique))), columns=index_unique)
+
+    with pytest.raises(InvalidIndexError):
+        _ = pd.concat([df_non_unique, df_unique], join=join)
+
+
+@pytest.mark.parametrize("index_maker", tm.index_subclass_makers_generator())
+@pytest.mark.xfail(reason="Not implemented")
```

Comment:
is there an issue for this? what is this case?

Reply:
I was expecting this to be allowed:

```python
import pandas as pd

result = pd.concat(
    [
        pd.Series(0, index=[0, 0, 1, 2]),
        pd.Series(1, index=[1, 2]),
    ],
    join="inner",
)
expected = pd.DataFrame({0: [0, 0], 1: [1, 1]}, index=[1, 2])

pd.testing.assert_frame_equal(result, expected)
```

Because the intersection of those indices is well defined. However, it turns out this does not work, and also doesn't work in 1.1.5. I sort of opened this issue here: #38773, but that was a more low-level issue.